sdk/src/main/java/io/dapr/client/domain/JobSchedule.java

```diff
@@ -0,0 +1,119 @@
+package io.dapr.client.domain;
+
+import java.time.Duration;
+
+/**
+ * Represents a job schedule using cron expressions or fixed intervals.
+ * This class provides various static methods to create schedules based on predefined periods
+ * (e.g., daily, weekly, monthly) or using custom cron expressions.
+ * Example usage:
+ * <pre>
+ * JobsSchedule schedule = JobsSchedule.daily();
+ * System.out.println(schedule.getExpression()); // Outputs: "0 0 0 * * *"
+ * </pre>
+ */
```

Comment:
I am wondering if this should be a simple cron expression holder and we should have a separate factory like `JobSchedules` that has the factory methods like `fromPeriod`, `early`, etc.

@cicoyle and @salaboy what are your thoughts.

Reply:
@artur-ciocanu my only concern was to complicate things when we integrate with Spring, so I would keep it lighter on the pure java side and then reuse all the Spring helpers to create Job expressions.. does that make sense?

Reply:
@salaboy my suggestion was to use `JobSchedule` as a simple class that wraps a `String` that is a CRON expression and we can have a helper factory like `JobScheduleFactory` that can have additional methods like `fromPeriod`, `fromString`, `early`, etc.

A CRON expression as a string is the lowest common denominator and it should be easy to integrate with Spring or any other framework.

Please let me know if I misunderstood your comment.